src/randgraphs.jl

```diff
@@ -148,6 +148,52 @@ function _try_creation(n::Int, k::Vector{Int}, rng::AbstractRNG)
     return edges
 end
 
+"""Creates a [Barabási–Albert model](https://en.wikipedia.org/wiki/Barab%C3%A1si%E2%80%93Albert_model) 
+random graph with `n` nodes is grown by attaching new nodes each with `m` edges that 
+are preferentially attached to existing nodes with high degree. Undirected graphs are
+created by default; use `is_directed=true` to override.
+"""
+function barabasi_albert(n::Integer, m::Integer; is_directed=false)
+	@assert(1<=m<n, "Barabási–Albert network must have 1 <= m < n")
+	if is_directed
+		g = DiGraph(n)
+	else
+		g = Graph(n)
+	end
+	# Target nodes for new edges
+	targets = collect(1:m)
```

Comment:
Do you need to `collect` these or can you use the range? (The range will save memory.)